<html>
<head>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
	<title>Network Science | Albert-László Barabás | Chapter 2</title>
	<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
	<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
	
  <script type="text/javascript" src="../javascripts/jquery.fixed.js"></script>

<script type="text/javascript" charset="utf-8">
    $(document).ready(function(){
      $('#toc-list').fixed({'top':'8'});
      $('#nav-bar-wrap').fixed({'top':'8'});
      $('span.c1').each(function(){ addStylesToCodeLines($(this)); });
      $('div.source-code').each(function(){ setRawCodeHeight($(this)); });
      $('a.toggle').click(function(){ toggleCodeDisplay($(this)); return false; });
    });
  </script>
  
	<link rel="stylesheet" href="../stylesheets/html.css" type="text/css">
    
    <link rel="stylesheet" href="../stylesheets/bjqs.css" type="text/css">
    <script src="../javascripts/bjqs-1.3.min.js"></script>


    <link href="../stylesheets/theme.default.css" rel="stylesheet">
	<script src="../javascripts/jquery.tablesorter.min.js"></script>
	<script src="../javascripts/jquery.tablesorter.widgets.min.js"></script>
	<script>
	$(function(){
		$('table').tablesorter({
			widgets        : ['zebra', 'columns'],
			usNumberFormat : false,
			sortReset      : true,
			sortRestart    : true
		});
	});
	</script>
    
    
    <script class="secret-source">
        jQuery(document).ready(function($) {
          
          $('#banner-slide').bjqs({
            animtype      : 'slide',
            height        : 413,
            width         : 800,
            responsive    : true,
          });
          
        });
      </script>
      
      <script class="secret-source2">
        jQuery(document).ready(function($) {
          
          $('#second-slide').bjqs({
            animtype      : 'slide',
            height        : 413,
            width         : 800,
            responsive    : true,
          });
          
        });
      </script>
      
      <script class="secret-source2">
        jQuery(document).ready(function($) {
          
          $('#third-slide').bjqs({
            animtype      : 'slide',
            height        : 413,
            width         : 800,
            responsive    : true,
          });
          
        });
      </script>
      

</head>
<body>

  <div id="navigator">
    <div id="navigator-inner">
      <div id='nav-bar-wrap'>
        <div id="mask"></div>
        <div id="nav-bar">
          <h1><a href="../index.html"> <strong>NETWORK SCIENCE</strong> BOOK</a></h1>
          <h2>Albert-László Barabási</h2>
          <a id="purchase-link" href="../chapter-3/ch3.html">CHAPTER 3</a> <a id="purchase-link" href="../chapter-1/ch1.html">CHAPTER 1</a>  
    <!--      <a id="purchase-link" href="../pdf/network_science_Ch2_2012_November.pdf">Download Chapter 2 as PDF</a>  -->
          
        </div>
      </div>
      <div id="toc-holder">
        <div id="toc-list">
          <ul>
            <li><a href="#bridges">2.1 The bridges of Königsberg</a></li> 
            <li><a href="#Networks-and-graphs">2.2 Networks and graphs</a></li>
            <li><a href="#Degree-average-degree-and-degree-distribution">2.3 Degree, average degree and degree distribution</a></li>
            <li><a href="#Real-networks-are-sparse">2.4 Real networks are sparse</a></li>
            <li><a href="#Adjacency-Matrix">2.5 Adjacency Matrix</a></li>
            <li><a href="#Weighted-and-unweighted-networks">2.6 Weighted and unweighted networks</a></li>
            <li><a href="#Bipartite-networks">2.7 Bipartite networks</a></li>
            <li><a href="#Paths-and-distances-in-networks">2.8 Paths and distances in networks</a></li>
            <li><a href="#Connectedness-and-Components">2.9 Connectedness and Components</a></li>
            <li><a href="#Clustering-coefficient">2.10 Clustering coefficient</a></li>
            <li><a href="#Case-study-and-summary">2.11 Case study and summary</a></li>
            <li><a href="#Global-clustering-coefficient">Appendix A: Global clustering coefficient</a></li>
            <li><a href="#Bibliography">2 Bibliography</a></li>
          </ul>
        </div>
      </div>
    </div>
  </div> <!--  fine  navigator //-->
  
  

  <div id="top">
    <div id="header">
      <h1><a href="##"><strong>NETWORK SCIENCE</strong> BOOK</a></h1>
      <h2>Albert-László Barabási</h2>
    </div>
  </div>

  <div id="middle">

  	<div id="container">
    
    
 <section>
    <a id="bridges"></a>
    <h2 id="#">Chapter 2 </br> </br>THE BRIDGES OF KÖNIGSBERG</h2>

<div class="image-container " >
	<video id="sampleMovie" width="100%" height="auto" preload controls>
    <source src="../imgs/chapter02/Konigsberg_bridges.m4v" />
	<source src="../imgs/chapter02/Konigsberg_bridges.ogv" />
	<source src="../imgs/chapter02/Konigsberg_bridges.webm" />
    </video>
<p class="captionBold">Image 2.1</p>
<p class="caption"><span class="imte">The bridges of Königsberg.</span></br>
From the contemporary map of Königsberg (now Kaliningrad, Russia) to Euler’s graph. The graph constructed by Euler consists of four nodes (<i>A, B, C, D</i>), each corresponding to a patch of land, and seven links, each corresponding to a bridge. Euler showed in 1736 that there is no continuous path that would cross seven the bridges while never crossing the same bridge twice. The people of Königsberg agreed with him, gave up their fruitless search and in 1875 they built a new bridge between <i>B</i> and <i>C</i>, increasing the number of links of these two nodes to four. Now only one node was left with an odd number of links and it became rather straightforward to find the desired path.</p>
</div>

<p>Few research fields can trace their birth to a single moment and place in history. Graph theory, the mathematical scaffold behind network science, can. Its roots go back to 1736 to Königsberg, the capital of Eastern Prussia and a thriving merchant city of its time. The trade supported by its busy fleet of ships allowed city officials to build seven bridges across the river Pregel that surrounded the town. Five of these connected the elegant island Kneiphof, caught between the two branches of the Pregel, to the mainland; two crossed the two branches of the river (<u>Image 2.1</u>). This peculiar arrangement gave birth to a contemporary puzzle: Can one walk across all seven bridges and never cross the same one twice? Despite many attempts, no one could find such path. The problem remained unsolved until 1735, when Leonard Euler, a Swiss born mathematician, offered a rigorous mathematical proof that such path does not exist.</p>
<p>Euler represented each of the four land areas separated by the river with letters A, B, C, and D. (<u>Image 2.1</u>). Next he connected with lines each piece of land that had a bridge between them. He thus built a <i>graph</i>, whose nodes were pieces of land and <i>links</i> were the bridges. Then Euler made a simple observation: if there is a path crossing all bridges, but never the same bridge twice, then nodes with odd number of links must be either the starting or the end point of this path. Indeed, if you arrive to a node with an odd number of links you may eventually have no unused link for you to leave it. A continuous path that goes through all bridges can have only one starting and one end point. Thus such a path cannot exist on a graph that has more than two nodes with an odd number of links. The Königsberg graph had three nodes with an odd number of links, B, C, and D, so no path could satisfy the problem.</p>
<p>Euler’s proof was the first time someone solved a mathematical problem by turning it into a graph. For us the proof has two important messages: the first is that some problems become simpler and more treatable if they are represented as a graph. The second is that the existence of the path does not depend on our ingenuity to find it. Rather, it is a property of the graph. Indeed, given the structure of the Königsberg graph, no matter how smart we are, we will never find the desired path. In other words, networks have properties hidden in their structure that limit or enhance their behavior. To fully understand how networks affect the properties of a system, we need to become familiar with graph theory, a branch of mathematics that grew out of Euler’s proof, offering a formalism that will be used throughout this book. </p>
</section>
<div class="pageb"></div> 

<section>
<a id="Networks-and-graphs"></br></br></a>
<h3 id="#">2.2 NETWORKS AND GRAPHS</h3>

<p>If we want to understand a complex system, we first need a map of its wiring diagram. A network is a catalog of a system’s components often called <b><em>nodes</em></b> or <b><em>vertices</em></b> and the direct interactions between them, called <b><em>links</em></b> or <b><em>edges</em></b> (Box 2.1).</br>
The network representation offers a common language to study systems that may differ greatly in nature, appearance, or scope. Indeed as shown in <u>Image 2.3</u>, three rather different systems have exactly the same network representation.</p>

<div class="tip">
  <h4 id="com.plex">Box 2.1 </br>Networks or graphs?</h4>
  <p>In the scientific literature the terms network and graph are used interchangeably. Yet, there is a subtle distinction between the two terminologies: the network, node, and link combination often refers to real systems: the WWW is a network of web pages connected by URLs; society is a network of individuals connected by family, friendship or professional ties; the metabolic network is the sum of all chemical reactions that take place in a cell. In contrast, we use the terms graph, vertex, and edge when we talk about the mathematical representation of these networks: we talk about the web graph, the social graph (a term made popular by Facebook), or the metabolic graph. Yet, this distinction is rarely made, so these two terminologies are often used as synonyms of each other.</p>
  
<table border="0">
<tr>
<th>Network Science</th>
<th>Graph Theory</th>
</tr>
<tr>
<td>network</td>
<td>graph</td>
</tr>
<tr>
<td>node</td>
<td>vertex</td>
</tr>
<tr>
<td>link</td>
<td>edge</td>
</tr>
</table>
</div>
<p><u>Image 2.3</u> also introduces two basic network parameters:</p>
<p><strong>Number of nodes</strong>, which we denote with <i>N</i>, representing the number of components in the system. We will often call <i>N</i> the <i>size of the network</i>.</p>
<p><strong>Number of links</strong>, which we denote with L, representing the total number of interactions between the nodes.</p>
<p>The networks shown in <u>Image 2.1</u> all have <i>N = 4</i> and <i>L = 4</i>. To distinguish the nodes, we label them <i>i = 1, 2, ..., N</i>. The links are rarely labeled, as they can be identified through the nodes they connect. For example, the (2, 4) link connects nodes 2 and 4.</p>

<div class="image-container" >
	<img src="../imgs/chapter02/ch02_03.png" alt="Real systems of quite different nature can have the same network representation." />	
    <p class="captionBold">Image 2.3</p>
    <p class="caption"><span class="imte">Real systems of quite different nature can have the same network representation.</span></br>
In the figure we show a small subset of (a) the Internet, where routers (specialized computers) are connected to each other; (b) the Hollywood actor network, where two actors are connected if they played in the same movie; (c) a protein-protein interaction network, where two proteins are connected if there is experimental evidence that they can bind to each other in the cell. While the nature of the nodes and the links differs widely, each network has the same graph representation, consisting of <i>N = 4</i> nodes and <i>L = 4</i> links, shown in (d).</p>
</div>

<p>The links of a network can be <i>directed</i> or <i>undirected</i>. Some systems have directed links, like the WWW, whose uniform resource locators (URL) point from one web document to the other, or phone calls, where one person calls the other. Other systems display undirected links, like romantic ties: if I date Janet, Janet also dates me, or transmission lines on the power grid, on which the electric current can flow in both directions.</p>

<p>A network is called <i>directed</i> (or digraph) if all of its links are directed or <i>undirected</i> if all of its links are undirected. Some networks simultaneously have directed and undirected links. For example in the metabolic network some reactions are reversible (i.e. bidirectional or undirected) and others are irreversible, taking place in only one direction (directed).</br>
Throughout this book we will use ten networks to illustrate the tools of network science. These networks, listed in <u>Table 2.1</u>, were selected having diversity in mind, spanning social systems (mobile call graph or email network), collaboration and affiliation networks (science collaboration network, Hollywood actor network), information systems (WWW), technological and infrastructural systems (Internet and power grid), biological systems (protein interaction and metabolic network), and reference networks (citations). They differ widely in their sizes, from as few as <i>N =1,039</i> nodes and <i>L = 5,802</i> links in the <i>E. coli</i> metabolism, to almost half million nodes and five million links in the citation network. They cover several of the areas where networks are actively applied, representing ‘canonical’ datasets, often used by researchers in the field of network science to illustrate key network properties. In the coming chapters we will discuss in detail the nature and the characteristics of each of these datasets, turning them into the guinea pigs of our journey to understand complex networks.</p>


	<div class="image-container" >
	<table class="tablesorter">
		<thead>
			<tr>
				<th>NETWORK NAME</th>
				<th>NODES</th>
				<th>LINKS</th>
				<th>DIRECTED/ UNDIRECTED</th>
                <th>N</th>
                <th>L</th>
                <th>‹K›</th>
			</tr>
		</thead>
		<tbody>
        <tr><td>Internet</td><td>routers</td><td>Internet Connections</td><td>Undirected</td><td>192,244</td><td>609,066</td><td>2.67</td></tr>
		<tr><td>WWW</td><td>webpages</td><td>links</td><td>Directed</td><td>325,729</td><td>1,497,134</td><td>4.60</td></tr>
        <tr><td>Power Grid</td><td>power plants, transformers</td><td>cables</td><td>Undirected</td><td>4,941</td><td>6,594</td><td>2.67</td></tr>
        <tr><td>Mobile-Phone Calls</td><td>subscribers</td><td>calls</td><td>Directed</td><td>36,595</td><td>91,826</td><td>2.51</td></tr>
        <tr><td>Email</td><td>email addresses</td><td>emails</td><td>Directed</td><td>57,194</td><td>103,731</td><td>1.81</td></tr>
        <tr><td>Science Collaboration</td><td>scientists</td><td>co-authorships</td><td>Undirected</td><td>23,133</td><td>186,936</td><td>16.16</td></tr>
        <tr><td>Actor Network</td><td>actors</td><td>co-acting</td><td>Undirected</td><td>212,250</td><td>3,054,278</td><td>28.78</td></tr>
        <tr><td>Citation Network</td><td>papers</td><td>citations</td><td>Directed</td><td>449,673</td><td>4,707,958</td><td>10.47</td></tr>
        <tr><td>E. coli Metabolism</td><td>metabolites</td><td>chemical reactions</td><td>Directed</td><td>1,039</td><td>5,802</td><td>5.84</td></tr>
        <tr><td>Yeast Protein Interactions</td><td>proteins</td><td>binding interactions</td><td>Undirected</td><td>2,018</td><td>2,930</td><td>2.90</td></tr>
        
		</tbody>
	</table>
<p class="captionBold">Table 2.1</p>
    <p class="caption"><span class="imte">Real systems of quite different nature can have the same network representation.</span></br>
The basic characteristics of the networks that we use throughout this book to illustrate the use of network science. This table lists the nature of their nodes and links, indicating if links are directed or undirected, the number of nodes <i>(N)</i> and links <i>(L)</i>, and the network’s average degree. For directed networks the average degree equals the average in- and out-degrees as <i>‹k› = ‹k<sub>in</sub>› = ‹k<sub>out</sub>›</i>.</p>
</div>

<div class="tip">
  <h4 id="com.plex">Box 2.2 </br>  Choosing the proper network representation.</h4>
  <p>The choices we make when we represent a complex system as a network will determine our ability to use network science successfully. For example, the way we define the links between two individuals dictates the nature of the questions we can explore:</p>
<div class="list">
<ul>
<li><p>By connecting individuals that regularly interact with each other in the context of their work, we obtain the <i>professional network</i>, that plays a key role in the success of a company or an institution, and it is of major interest to organizational research.</p></li>
<li><p>By linking friends to each other, we obtain the <i>friendship network</i>, that plays an important role in the spread of ideas, products and habits and is of major interest to sociology, marketing and health sciences.</p></li>
<li><p>By connecting individuals that have an intimate relationship, we obtain the <i>sexual network</i>, of key importance for the spread of sexually transmitted diseases, like AIDS, and of major interest for epidemiology.</p></li>
<li><p>By using phone and email records to connect individuals that call or email each other, we obtain the <i>acquaintance network</i>, capturing a mixture of professional, friendship or intimate links, of importance to communications and marketing.</p></li>
</ul>
</div>

<p>While many links in these four networks overlap (some coworkers may be friends or may have an intimate relationship), these networks are not identical. Other networks may be valid from a graph theoretic perspective, but may have little practical utility. For example, by linking all individuals with the same first name, Johns with Johns and Marys with Marys, we do obtain a well-defined network, yet its utility is questionable. Hence in order to apply network theory to a system, careful considerations must precede our choice of nodes and links, ensuring their significance to the problem we wish to explore.</p>
</div>

</section>
<div class="pageb"></div>


<section>
<a id="Degree-average-degree-and-degree-distribution"></br></br></a>
<h3 id="#">2.3 DEGREE, AVERAGE DEGREE, AND DEGREE DISTRIBUTION</h3>

<p>A key property of each node is its <i>degree</i>, representing the number of links it has to other nodes. The degree can represent the number of mobile phone contacts an individual has in the call graph (i.e. the number of different individuals the person has talked to), or the number of citations a research paper gets in the citation network.</br>
We denote with <i>k<sub>i</sub></i> the degree of the <i>i<sup>th</sup></i> node in the network. For example, for the undirected networks shown in <u>Image 2.3</u> we have <i>k<sub>1</sub>=2, k<sub>2</sub>=3, k<sub>3</sub>=2, k<sub>4</sub>=1</i>.</br>
In an undirected network total number of links, <i>L</i>, can be expressed as the sum of the node degrees:</p>

<h4> \[ L = \frac{1}{2}\sum\limits_{i = 1}^N {{k_i}}  \hspace{20 mm} (1) \]  </h4> 

<p>Here the 1/2 factor corrects for the fact that in the sum (1)
each link is counted twice. For example, the link connecting the nodes 2 and 4 in <u>Image 2.3</u> will be counted once in the degree of node <i>1 (k<sub>2</sub> = 3)</i> and once in the degree of node <i>4 (k<sub>4</sub> = 1)</i>.</p>

<div class="tip">
  <h4 id="com.plex">Box 2.3 </br>Brief statistics review.</h4>
  <p>The average, the standard deviation, and the distribution of random variables will play a key role throughout this book.</br>
For a sample of <i>N</i> values <i>x<sub>1</sub>, ... , x<sub>N</sub></i> we have:</p>

<p>Average (mean value):</p>
<h4>  \[\left\langle x \right\rangle  = \frac{{{x_1} + {x_2} + ... + {x_N}}}{N} = \frac{1}{N}\sum\limits_{i = 1}^N {{x_i}}   \hspace{20 mm} (2) \]  </h4> 

<p><i>n<sup>th</sup></i> moment:</p>

<h4>\[\left\langle x \right\rangle  = \frac{{x_1^n + x_2^n + ... + x_N^n}}{N} = \frac{1}{N}\sum\limits_{i = 1}^N {x_i^n}    \hspace{20 mm} (3) \]</h4>

<p>Standard deviation (fluctuations around the average):</p>
<h4>\[{\sigma _X} = \sqrt {\frac{1}{N}\sum\limits_{i = 1}^N {{{\left( {{x_i} - \left\langle x \right\rangle } \right)}^2}} } \hspace{20 mm} (4) \]</h4>

<p>Distribution of <i>x</i> (probability that a randomly chosen value is a):</p>
<h4>\[p = \frac{1}{N}\sum\limits_i {{\delta _{x,{x_i}}}} \hspace{20 mm} (5) \]</h4>

<p>which yields</p>
<h4>\[\sum\limits_i {{p_x}}  = 1\left( {\int {{p_x}dx = 1} } \right) \hspace{20 mm} (6) \]</h4>

</div>

<p>An important property of a network is its <i>average degree</i>, which for an undirected network is</p>
<h4>\[\left\langle k \right\rangle  \equiv \frac{1}{N}\sum\limits_{i = 1}^N {{k_i}}  = \frac{{2L}}{N}\hspace{20 mm} (7) \]</h4>

<p>In directed networks we distinguish between <i>incoming degree, k<sub>i</sub><sup>in</sup></i>, representing the number of links that point node <i>i</i>, and <i>outgoing degree, k<sub>i</sub><sup>out</sup></i> , representing the number of links that point from the node <i>i</i> to other nodes and the <i>total degree, k<sub>i</sub></i> , given by</p>

<h4>\[{k_i} = k_i^{in} + k_i^{out}   \hspace{20 mm} (8) \]</h4>

<p>For example, on the WWW the number of pages a given document points to represents its outgoing degree, <i>k<sub>out</sub></i>, and the number of other documents that point to it represents its incoming degree, <i>k<sub>in</sub></i>.</p>

<p>The total number of links in a directed network is</p>
<h4>\[l = \sum\limits_{i = 1}^N {k_i^{in}}  = \sum\limits_{i = 1}^N {k_i^{out}} \hspace{20 mm} (9) \]</h4>

<p>The 1/2 factor in Eq. (1), is absent above, as for directed networks the two sums in (9) separately count the outgoing and the incoming degrees.</p>

<p>The average degree of a directed network is</p>
<h4>\[ \left( {{k^{in}}} \right) = \frac{1}{N}\sum\limits_{i = 1}^N {k_i^{in}}  = \left( {{k^{out}}} \right) = \frac{1}{N}\sum\limits_{i = 1}^N {k_i^{out}}  = \frac{L}{N} \hspace{20 mm} (10) \]</h4>

<div class="image-container " >
	<img src="../imgs/chapter02/ch02_04a.jpg" alt="Degree distribution." />	
    <p class="captionBold">Image 2.4a</p>
    <p class="caption"><span class="imte">Degree distribution.</span></br>
The degree distribution is defined as the <i>p<sub>k</sub> = N<sub>k</sub> /N</i> ratio, where <i>N<sub>k</sub></i> denotes the number of <i>k-degree</i> nodes in a network. For the network in (a) we have <i>N = 4</i> and <i>p<sub>1</sub> = 1/4</i> (one of the four nodes has degree <i>k<sub>1</sub> = 1</i>), <i>p<sub>2</sub> = 1/2</i> (two nodes have k<sub>3</sub> = k<sub>4</sub> = 2), and <i>p<sub>3</sub> = 1/4</i> (as k<sub>2</sub> = 3). As we lack nodes with degree <i>k > 3</i>, <i>p<sub>k</sub> = 0</i> for any <i>k > 3</i>. Panel (b) shows the degree distribution of a one dimensional lattice. As each node has the same degree <i>k = 2</i>, the degree distribution is a Kronecker’s delta function <i>p<sub>k</sub> = &#948;(k - 2).</i></p>
</div>

<div class="image-container " >
	<img src="../imgs/chapter02/ch02_04b.jpg" alt="Degree distribution." />	
    <p class="captionBold">Image 2.4b</p>
    <p class="caption"><span class="imte">Degree distribution.</span></br>
In many real networks, the node degree can vary considerably. For example, as the degree distribution (a) indicates, the degrees of the proteins in the protein interaction network shown in (b) vary between <i>k=0</i> (isolated nodes) and <i>k=92</i>, which is the degree of the largest node, called a hub. There are also wide differences in the number of nodes with different degrees: as (a) shows, almost half of the nodes have degree one (i.e. <i>p<sub>1</sub>=0.48</i>), while there is only one copy of the biggest node, hence <i>p<sub>92</sub> = 1/ N=0.0005.</i> (c) The degree distribution is often shown on a so-called log-log plot, in which we either plot log p<sub>k</sub> in function of <i>log k</i>, or, as we did in (c), we use logarithmic axes.</p>
</div>

<p>The <i>degree distribution, p<sub>k</sub></i>, provides the probability that a randomly selected node in the network has degree <i>k</i>. Since <i>p<sub>k</sub></i> is a probability, it must be normalized, i.e. \[ \sum\limits_{k = 1}^\infty  {{p_k}}  = 1 \] . For a fixed network of <i>N</i> nodes the degree distribution is the normalized histogram (see Gallery 2.1), \[  {p_k} = \frac{{{N_k}}}{N}  \] , where <i>N<sub>k</sub></i> is the number of degree <i>k</i> nodes. Hence the number of degree <i>k</i> nodes can be obtained from the degree distribution as <i>N<sub>k</sub></i> = <i>N<sub>pk</sub></i>.</p>



<p>The degree distribution has taken a central role in network theory following the discovery of scale-free networks (Barabási & Albert, 1999). Another reason for its importance is that the calculation of most network properties requires us to know <i>p<sub>k</sub></i>. For example, the average degree of a network can be written as \[  \left\langle k \right\rangle  = \sum\limits_{k = 0}^\infty  {k{p_k}}  \]</p>

<p>We will see in the coming chapters that the precise functional form of <i>p<sub>k</sub></i> determines many network phenomena, from network robustness to the spread of viruses.</p>

</section>
<div class="pageb"></div>


<section>
<a id="Real-networks-are-sparse"></br></br></a>
<h3 id="#">2.4 REAL NETWORKS ARE SPARSE</h3>

<div class="image-container half-width-right" >
	<video id="sampleMovie" width="100%" height="auto"  preload controls autoplay loop>
    <source src="../imgs/chapter02/complete_graph.m4v" />
	<source src="../imgs/chapter02/complete_graph.ogv" />
	<source src="../imgs/chapter02/complete_graph.webm" />
    </video>	
    <p class="captionBold">Image 2.5</p>
    <p class="caption"><span class="imte">Complete graph.</span></br>
The figure shows a complete graph with <i>N = 16</i> nodes and <i>L<sub>max</sub> = 120</i> links, as predicted by Eq. (11). The adjacency matrix of a complete graph is <i>A<sub>ij</sub> = 1</i> for all <i>i, j = 1, ....N</i> and <i>A<sub>ii</sub> = 0</i>. The average degree of a complete graph is <i>‹k› = N - 1</i>.</p>
</div>

<p>In real networks the number of nodes (<i>N</i>) and links (<i>L</i>) can vary widely. For example, the neural network of the worm <i>C. elegans</i>, the only fully mapped brain of a living organism, has <i>297</i> neurons (nodes) and <i>2,345</i> synapses (links), while a human brain is estimated to have about a hundred billion (<i>10<sup>11</sup></i>) neurons, each with an average of 7,000 synaptic connections. The genetic network of a human cell has about 20,000 genes as nodes; the social network consists of seven billion individuals (<i>N &#8771; 7×10<sup>9</sup></i>) and the WWW is estimated to have over a trillion webpages (<i>N>10<sup>12</sup></i>). These wide differences in size are noticeable in <u>Table 2.1</u> where we list <i>N</i> and <i>L</i> for several network maps. Some of these maps offer a complete wiring diagram of the system they describe (like the actor network or the <i>E. Coli</i> metabolism), others are only samples, representing a subset of a real system’s nodes (WWW, mobile call graph).</p>



<p><u>Table 2.1</u> indicates that the number of links also varies widely. In a network of <i>N</i> nodes the number of links is between <i>L = 0</i> and <i>L<sub>max</sub></i>, where <i>L<sub>max</sub></i> is the total number of links present in a complete graph (<u>Image 2.5</u>), a graph in which each node is connected to all other nodes. In real networks L is much smaller than <i>L<sub>max</sub></i>, indicating that real networks are sparse. For example, the WWW graph in <u>Table 2.1</u> has about 1.5 million links. Yet, if the WWW were to be a complete graph, this sample should have <i>L<sub>max</sub> ≈ 10<sup>12</sup></i> links according to (11).</p>

<h4>\[{L_{\max }} = \left( {\begin{array}{*{20}{c}}N\\2\end{array}} \right) = \frac{{N\left( {N - 1} \right)}}{2} \hspace{20 mm} (11) \]</h4>

<p>Therefore, the web graph has only a <i>10<sup>-6</sup></i> fraction of the links it could have, making it a sparse network. In fact each network in <u>Table 2.1</u> has only a tiny fraction of the links it could have according to (11). As we will see later sparseness has important consequences on the way we explore and store real networks.</p>


</section>
<div class="pageb"></div>


<section>
<a id="Adjacency-Matrix"></br></br></a>
<h3 id="#">2.5 ADJACENCY MATRIX</h3>

<p>A full description of a network requires us to keep track of its links. The simplest way to achieve this is to provide a complete list of the links. For example, the network of <u>Image 2.1</u> is uniquely described by the list of its four <i>(i, j)</i> links: <i>{(1, 2), (1, 3), (2, 3), (2, 4)}</i>.</p>

<p>For mathematical purposes we often represent a network through its adjacency matrix. The adjacency matrix of a directed network of <i>N</i> nodes has <i>N</i> rows and <i>N</i> columns, its elements being:</p>

<p><i>A<sub>ij</sub> = 1</i> if there is a link pointing from node <i>j</i> to node <i>i</i></p>

<p><i>A<sub>ij</sub> = 0</i> if nodes <i>i</i> and <i>j</i> are not connected to each other.</p>

<p>The adjacency matrix of an undirected network has two entries for each link, e.g. <i>link (1,2)</i> is represented as <i>A<sub>12</sub> =1 and A<sub>21</sub> =1</i>. Hence the adjacency matrix of an undirected network is symmetric, i.e. <i>A<sub>ij</sub> = A<sub>ij</sub></i> (<u>Image 2.7</u>).</p>

<p>The degree <i>k<sub>i</sub></i> of node i can be directly obtained from the elements of the adjacency matrix. For undirected networks a node’s degree is a sum over either the rows or the columns of the matrix, i.e.</p>

<h4>\[ {k_i} = \sum\limits_{j = 1}^N {{A_{ij}}}  = \sum\limits_{i = 1}^N {{A_{ij}}}   \hspace{20 mm} (12)\]</h4>

<p>For directed networks the sums over the adjacency matrix’rows and columns provide the incoming and outgoing degrees, respectively</p>

<h4>\[  k_i^{in} = \sum\limits_{j = 1}^N {{A_{ij}}} \hspace{20 mm}   k_i^{out} = \sum\limits_{j = 1}^N {{A_{ij}}} \hspace{20 mm} (13) \]</h4>

<p>Given that in an undirected network the number of outgoing links equals the number of incoming links, we have</p>

<h4>\[2L = \sum\limits_{i = 1}^N {k_i^{in} = } \sum\limits_{j = 1}^N {k_j^{out} = } \sum\limits_{ij}^N {{A_{ij}}} \hspace{20 mm} (14) \]</h4>

<div class="image-container half-width-right" >
	<img src="../imgs/chapter02/ch02_06.png" alt="The adjacency matrix is typically sparse." />	
    <p class="captionBold">Image 2.6</p>
    <p class="caption"><span class="imte">The adjacency matrix is typically sparse.</span></br>
The adjacency matrix of the yeast protein-protein interaction network, consisting of <i>2,018</i> nodes, each representing a yeast protein (<u>Table 2.1</u>).
A dot is placed on each spot of the adjacent matrix for which <i>A<sub>ij</sub> = 1</i>, indicating the presence of an interaction. There are no dots for <i>A<sub>ij</sub> = 0</i>. The small fraction of dots underlines the sparse nature of the protein-protein interaction network.</p>
</div>

<p>The number of nonzero elements of the adjacency matrix is <i>2L</i>, or twice the number of links. Indeed, an undirected link connecting nodes <i>i</i> and <i>j</i> appears in two entries:</br>
<i>A<sub>ij</sub> = 1</i>, a link pointing from node <i>j</i> to node <i>i</i>, and <i>A<sub>ji</sub> = 1</i>, and a link pointing from <i>i</i> to <i>j</i> (<u>Image 2.7</u>).</p>

<p>The sparsity of real networks implies that the adjacency matrices are also sparse. Indeed, a complete network has <i>A<sub>ij</sub> = 1</i>, for all <i>(i, j)</i>, i.e. each of its matrix elements are equal to one. In contrast in real networks only a tiny fraction of the matrix elements are nonzero. This is illustrated in <u>Image 2.6</u>, where we show the adjacency matrix of the protein-protein interaction network listed in <u>Table 2.1</u>. One can see that the matrix appears nearly empty. One immediate consequence of the sparseness is that when we store a large network in our computer, it is better to store only the list of links (i.e. elements for which <i>A<sub>ij</sub> ≠ 0</i>), rather than full adjacency matrix, as an overwhelming fraction of <i>A<sub>ij</sub></i> elements are zero.</p>

<div class="image-container" >
	
   <h4>\[ Adjacency matrix \]</h4>
    <h4>\[{A_{ij}} = \left( {\begin{array}{*{20}{c}}
{{A_{11}}}&{{A_{11}}}&{{A_{13}}}&{{A_{14}}}\\
{{A_{21}}}&{{A_{22}}}&{{A_{23}}}&{{A_{24}}}\\
{{A_{31}}}&{{A_{32}}}&{{A_{33}}}&{{A_{34}}}\\
{{A_{41}}}&{{A_{42}}}&{{A_{43}}}&{{A_{44}}}
\end{array}} \right)\]
</h4>
    
    <h4>\[Undirected network \hspace{40 mm}  Directed network \]</h4>
    <img src="../imgs/chapter02/ch02_07.jpg"  />
    <h4>\[{A_{ij}} = \left( {\begin{array}{*{20}{c}}
0&1&1&0\\
1&0&1&1\\
1&1&0&0\\
0&1&0&0
\end{array}} \right)  \hspace{30 mm}  {A_{ij}} = \left( {\begin{array}{*{20}{c}}
0&0&0&0\\
1&0&1&0\\
0&0&0&0\\
0&1&0&0
\end{array}} \right)
 \]
</h4>

<h4>\[{k_2} = \sum\limits_{j = 1}^4 {{A_{2j}}}  = \sum\limits_{i = 1}^4 {{A_{i2}}}  = 3  \hspace{30 mm}  k_2^{in} = \sum\limits_{j = 1}^4 {{A_{2j}}}  = 2 \]</h4>
<h4>\[ \hspace{110 mm} k_2^{out} = \sum\limits_{i = 1}^4 {{A_{i2}}}  = 1\]</h4>
<h4>\[  {A_{ij}} = {A_{ji}}   \hspace{10 mm}  {A_{ii}} = 0 \hspace{45 mm}  {A_{ij}} \ne {A_{ji}}  \hspace{10 mm}  {A_{ii}} = 0   \]</h4>
<h4>\[L = \frac{1}{2}\sum\limits_{i,j = 1}^N {{A_{ij}}} \hspace{10 mm}  \left\langle k \right\rangle  = \frac{{2L}}{N}   \hspace{20 mm}  L = \sum\limits_{i,j = 1}^N {{A_{ij}}} \hspace{10 mm}  \left\langle {{k^{in}}} \right\rangle  = \left\langle {{k^{out}}} \right\rangle  = \frac{L}{N}     \]</h4>
    
    
    <p class="captionBold">Image 2.7</p>
    <p class="caption"><span class="imte">The adjacency matrix.</span></br>
Top: The elements of the adjacency matrix. The adjacency matrix of a directed (left column) and an undirected (right column) network. The figure highlights the fact that the degree of a node (in this case node 2) can be expressed as the sum over the appropriate column or row of the adjacency matrix. It also shows a few basic network characteristics, like the total number of links, (<i>L</i>), and average degree, (<i>‹k›</i>), expressed in terms of the elements of the adjacency matrix.</p>
</div>

</section>
<div class="pageb"></div>


<section>
<a id="Weighted-and-unweighted-networks"></br></br></a>
<h3 id="#">2.6 WEIGHTED AND UNWEIGHTED NETWORKS</h3>

<p>So far we discussed only networks for which all links have the same weight, i.e. <i>A<sub>ij</sub> = 1</i>. Yet, in many applications we need to study weighted networks, where each link <i>(i, j)</i> has a unique weight <i>w<sub>ij</sub></i> . In mobile call networks the weight can represent the total number of minutes two mobile phone users talk with each other on the phone; on the power grid the weight is the amount of current flowing through a transmission line.</p>

<p>For weighted networks the elements of the adjacency matrix carry the weight of the link</p>

<h4>\[{A_{ij}} = {w_{ij}} \hspace{20 mm} (15) \]</h4>

<p>Most networks of scientific interest are weighted, but we can not always measure the appropriate weights, hence we often approximate these networks as unweighted. In this book we predominantly focus on unweighted networks, but we will devote a separate chapter to network characteristics that are unique to weighted networks.</p>

<div class="tip">
  <h4 id="com.plex">Box 2.4 </br>  The value of a network: Metcalfe’s Law.</h4>
  <p>Metcalfe’s law states that the value of a network is proportional to the square of the number of its nodes, i.e. <i>N<sup>2</sup></i>. Formulated around 1980 in the context of communication devices by Robert M. Metcalfe (Gilder, 1993), the idea behind Metcalfe’s law is that the more individuals use a network, the more valuable it becomes. Indeed, the more of your friends use email, the more valuable it is to you as well, as the more individuals you can communicate with.</p>
  
<div class="image-container half-width-right" >
	<img src="../imgs/chapter02/ch02_08.jpg" alt="The adjacency matrix is typically sparse." />	
    <p class="captionBold">Image 2.8</p>
    <p class="caption"><span class="imte">Metcalfe’s law.</span></br>
According to Metcalfe’s law the cost of network based services and products increases linearly with the number of <i>nodes</i> (users or devices) while the benefits or income is driven by the number of <i>links L</i> the max technology makes possible, growing like <i>N<sup>2</sup></i>. Hence once the number of devices exceeds some <cite>"critical mass crossover"</cite>, the technology becomes profitable.</p>
</div>
  
<p>During the Internet boom of the late 1990s Metcalfe’s law was frequently used to offer a quantitative valuation for Internet companies, supporting a <cite>“build it and they will come” mentality (Briscoe et al., 2006)</cite>. It suggested that the value of a service is proportional to the square of the number of its users, in contrast with the cost that grows only linearly. Hence if the service attracts sufficient number of users, it will inevitably become profitable, as <i>N<sup>2</sup></i> will surpass <i>N</i> at some sufficiently large <i>N</i>. Hence Metcalfe’s Law offered credibility to growth over profits, fueling the Internet bubble of 2001.</p>

<p>Metcalfe’s law is based on <u>Eq. (11)</u>, telling us that if all links of a communication network with <i>N</i> nodes are equally valuable, the total value of the network is proportional to <i>N(N - 1)/2</i>, that is, roughly, <i>N<sup>2</sup></i>. If a network has <i>N = 10</i> members, there are <i>L<sub>max</sub> = 45</i> different possible connections between them. If the network doubles in size to <i>N = 20</i>, the number of connections doesn’t merely double but roughly quadruples to 190, a phenomenon called <i>network externality</i> in economics.</p>

<p>Two issues limit the validity of Metcalfe’s law: <br>
(i) most real networks are sparse, which means that only a very small fraction of the links are present. Hence the total value of the network will not grow like <i>N<sup>2</sup></i>, but the growth is often only linear in <i>N</i>.<br>
(ii) As the links have weights, not all links are of equal value; some links are used heavily while the vast majority of links are rarely utilized.</p>
  
</div>

</section>
<div class="pageb"></div>


<section>
<a id="Bipartite-networks"></br></br></a>
<h3 id="#">2.7 BIPARTITE NETWORKS</h3>

<div class="image-container" >
	<img src="../imgs/chapter02/ch02_09.jpg" alt="Bipartite network." />	
    <p class="captionBold">Image 2.9a - 2.9b</p>
    <p class="caption"><span class="imte">Bipartite network - Image 2.9a.</span></br>
In a bipartite network we have two sets of nodes, <i>U</i> and <i>V</i>, so that nodes in the <i>U-set</i> connect directly only to nodes in the <i>V-set</i>. Hence there are no direct <i>U-U</i> or <i>V-V</i> links. The figure also shows the two projections we can generate from any bipartite network. Projection <i>U</i> is obtained by connecting two <i>U-nodes</i> to each other if they link to the same <i>V-node</i> in the bipartite representation. Projection <i>V</i> is obtained by connecting two <i>V-nodes</i> to each other if they link to the same <i>U-node</i> in the bipartite network.</p>

    <p class="caption"><span class="imte">Bipartite network - Image 2.9b.</span></br>
The <i>human diseaseome</i> is a bipartite network, whose nodes are diseases (<i>U</i>) and genes (<i>V</i>), in which a disease is connected to a gene if mutations in that gene are known to affect the particular disease [4]. One projection of the diseaseome is the <i>gene network</i>, whose nodes are genes, two genes being connected if they are associated with the same disease. The second projection is the disease network, whose nodes are diseases, two diseases being connected if the same genes are associated with them, indicating that the two diseases have common genetic origins. The figure shows a subset of the diseaseome, focusing on cancers. The full human diseaseome map, connecting 1,283 disorders via 1,777 shared disease genes. (After [4])</p>
</div>

<p>A bipartite graph (or bigraph) is a network whose nodes can be divided into two disjoint sets <i>U</i> and <i>V</i> such that each link connects a <i>U-node</i> to a <i>V-node</i>. In other words, if we color the <i>U-nodes</i> yellow and the <i>V-nodes</i> green, then each link must connect nodes of different colors (<u>Image 2.9a/b</u>).</p>

<div class="image-container half-width-right" >
	<img src="../imgs/chapter02/ch02_10a.jpg" alt="Tripartite network." />	
    <p class="captionBold">Image 2.10a</p>
    <p class="caption"><span class="imte">Tripartite network.</span></br>
The tripartite recipe-ingredient-compound network, in which one set of nodes are recipes, like Chicken Marsala, the second set corresponds to the ingredients each recipe has (like flour, sage, chicken, wine, and butter for Chicken Marsala), and the third set captures the flavor compounds, or chemicals that contribute to the taste of a particular ingredient.</p>
</div>

<p>We can generate two projections for each bipartite network. The first projection connects two <i>U-nodes</i> to each other by a link if they are linked to the same <i>V-node</i> in the bipartite representation; the second projection connects the <i>V-nodes</i> to each other by a link of they connect to the same <i>U-node</i>.</p>

<p>In network theory we encounter numerous bipartite networks. A well-known example is the Hollywood actor network, in which one set of nodes corresponds to movies (<i>U</i>), and the other to actors (<i>V</i>), a movie being connected to an actor if the actor plays in that movie. In this network one projection corresponds to the actor network, in which two nodes are connected to each other if they played in the same movie; this is the network characterized in <u>Table 2.1</u>. The other projection is the movie network, in which wo movies are connected if they share at least one actor in their cast. Another example of bipartite network emerges in medicine, connecting diseases to the genes whose effects can cause or influence the corresponding disease (<u>Image 2.9a/b</u>). Finally, one can also define multipartite networks, like the tripartite recipe-ingredient-compound network described in <u>Image 2.10 a/b</u>.</p>

<div class="image-container" >
<script src="http://zoom.it/loA6.js?width=auto&height=400px"></script>
<p class="captionBold">Image 2.10b</p>
    <p class="caption"><span class="imte">Tripartite network.</span></br>
A projection of the tripartite network, resulting in the ingredient network, often called the flavor network. Each node denotes an ingredient; the node color indicating the food category and node size reflects the ingredient prevalence in recipes. Two ingredients are connected if they share a significant number of flavor compounds, link thickness representing the number of shared compounds between the two ingredients (After [12]).</p>
</div>

</section>
<div class="pageb"></div>


<section>
<a id="Paths-and-distances-in-networks"></br></br></a>
<h3 id="#">2.8 PATHS AND DISTANCES IN NETWORKS</h3>

<p>In physical systems the components are characterized by obvious distances, like the distance between two atoms in a crystal, or between two galaxies in the universe. In networks distance is a challenging concept. Indeed, what is the distance between two webpages on the WWW, or two individuals who may or may not know each other? The physical distance is not relevant here: two webpages linked to each other could be sitting on computers on the opposite sides of the globe and two individuals, living in the same building, may not know each other. In networks physical distance is replaced by <i>path length</i>. A <i>path</i> is a route that runs along the links of the network, its length representing the number of links the path contains. A path can intersect itself and pass through the same link repeatedly (<u>Image 2.5</u>). In network science paths play a central role, hence next we discuss some of their most important properties, many more being summarized in <u>Gallery 2.4</u>.</p>

<div class="image-container half-width-right" >
	<img src="../imgs/chapter02/ch02_11.jpg" alt="The adjacency matrix is typically sparse." />	
    <p class="captionBold">Image 2.11</p>
    <p class="caption"><span class="imte">The adjacency matrix is typically sparse.</span></br>
(a) A path between nodes <i>i<sub>0</sub></i> and <i>i<sub>n</sub></i> is an ordered list of <i>n</i> links <i>P<sub>d</sub> = {(i<sub>0</sub>, i<sub>1</sub>), (i<sub>1</sub>, i<sub>2</sub>), (i<sub>2</sub>, i<sub>3</sub>), ... ,(i<sub>n-1</sub>, i<sub>n</sub>),}</i>.The length of this path is <i>d</i>. The path shown in (a) follows the route <i>1→2→5→4→2→5→7</i>, hence its length is <i>n = 6</i>.<br>
(b) The shortest paths between nodes 1 and 7, representing the distance <i>d<sub>17</sub></i> , is the path with the fewest number of links that connect nodes 1 and 7. There can be multiple paths of the same length, as illustrated by the two paths shown in different colors. The network diameter is the largest distance in the network, being <i>d<sub>max</sub> = 3</i> here.</p>
</div>

<p><strong>Shortest Path</strong> (or geodesic path) between nodes <i>i</i> and <i>j</i> is the path with fewest number of links (<u>Image 2.5</u>). The shortest path is often called the distance between nodes <i>i</i> and <i>j</i>, and is denoted by <i>d<sub>ij</sub></i>, or simply <i>d</i>. We can often find multiple shortest paths of the same length d between a pair of nodes (<u>Image 2.5</u>). The shortest path never contains loops or intersects itself.</p>

<p>In an undirected network <i>d<sub>ij</sub> = d<sub>ji</sub></i>, i.e. the distance between node <i>i</i> and <i>j</i> is the same as the distance between node <i>j</i> and <i>i</i>. In a directed network often <i>d<sub>ij</sub> ≠d<sub>ji</sub></i>. Furthermore, in a directed network the existence of a path from node <i>i</i> to node <i>j</i> does not guarantee the existence of a path from <i>j</i> to <i>i</i>.</p>

<p>In real networks we frequently need to determine the distance between two nodes. For a small network, like the one shown in <u>Image 2.5</u>, this is an easy task. For a network of millions of nodes finding the shortest path between two nodes can be rather time consuming. The length of the shortest path and the number of such paths can be formally obtained from the adjacency matrix (<u>Box 2.5</u>). In practice we most often use the breadth first search (BFS) algorithm discussed in <u>Box 2.6</u> and <u>Gallery 2.5</u> to measure the distance between two nodes.</p>

<div class="tip">
  <h4 id="com.plex">Box 2.5</br>Number of shortest paths between two nodes.</h4>
  <p>The number of shortest paths, <i>N<sub>ij</sub></i> , between nodes <i>i</i> and <i>j</i> and the distance <i>d<sub>ij</sub></i> between them can be determined directly from the adjacency matrix, <i>A<sub>ij</sub></i> .</p>
  
  <ul>
<li><p> <i>d<sub>ij</sub> = 1</i>: If there is a link between <i>i</i> and <i>j</i>, then <i>A<sub>ij</sub> = 1</i> (<i>A<sub>ij</sub> = 0</i> otherwise).</p></li>
<li><p> <i>d<sub>ij</sub> = 2</i>: If there is a path of length two between <i>i</i> and <i>j</i>, then the product of <i>d</i> elements <i>A<sub>ik</sub> A<sub>kj</sub> = 1</i> (<i>A<sub>ik</sub> A<sub>kj</sub> = 0</i> otherwise). The number of <i>d<sub>ij</sub> = 2</i> paths between <i>i</i> and <i>j</i> is</p>       

<p>\[N_{ij}^{\left( 2 \right)} = \sum\limits_{k = 1}^N {{A_{ik}}} {A_{kj}} = {\left[ {{A^2}} \right]_{ij}} \hspace{20 mm} (16) \]</p>

<p>where <i>[...]ij</i> denotes the <i>(ij)<sup>th</sup></i> element of a matrix.</p></li>
<li><p> <i>d<sub>ij</sub> = d</i>: If there is a path of length <i>d</i> between <i>i</i> and <i>j</i>, then <i>A<sub>ik</sub> ... A<sub>lj</sub> = 1</i> (<i>A<sub>ik</sub> ... A<sub>lj</sub> = 0</i> otherwise). The number of paths of length <i>d</i> between <i>i</i> and <i>j</i> is</p>

<p>\[N_{ij}^{\left( d \right)} = {\left[ {{A^d}} \right]_{ij}}  \hspace{20 mm} (17)  \]</p>

</li>
</ul>
  
  <p>Equation (17) holds for both directed and undirected networks and can be generalized to multigraphs as well. The distance between nodes <i>i</i> and <i>j</i> is the path with the smallest <i>d</i> for which <i>N<sub>ij</sub><sup>(d)</sup> > 0</i>. Despite the mathematical elegancy of Eq. (17), faced with a large network, it is more efficient to use the breadth-first-search algorithm described in <u>Box 2.6</u>.</p>
</div>

<p><strong>Network diameter:</strong> the diameter of a network, denoted by <i>d<sub>max</sub></i>, is the maximal shortest path in the network. In other words, it is the largest distance recorded between any pair of nodes. One can verify that the diameter of the network shown in <u>Image 2.5</u> is <i>d = 3</i>. For larger graphs the max￼diameter can also be determined using the breadth first search algorithm (<u>Box 2.6</u>).</p>

<p><strong>Average path length</strong>, denoted by <i>‹d›</i>, is the average distance between all pairs of nodes in the network. For a directed network of <i>N</i> nodes, <i>‹d›</i> is given by</p>

<h4>\[\left\langle d \right\rangle  = \frac{1}{{N\left( {N - 1} \right)}}\sum\limits_{i,j = 1,N} {{d_{i,j}}} \hspace{20 mm} (18)  \]</h4>

<p>For an undirected network we need to multiply the r.h.s. of Eq. (18) by two.</p>

<p>We can use the BFS algorithm to determine the average path length for a large network. For this we first determine the distance between a node and all other nodes in the network using the algorithm described in <u>Box 2.6</u>. We then determine the shortest path between a second node and all other nodes but the first one, a procedure that we repeat for all nodes. The sum of these shortest paths divided by <i>L<sub>max</sub></i> provides the average path length.</p>

<div class="tip">
  <h4 id="com.plex">Box 2.6</br>Finding the shortest path: breath first search.</h4>
  <p>BFS is one of the most frequently used algorithms in network science. Similar to throwing a pebble in a pond and watching the ripples spread from the center, we start from a node and label its neighbors, then the neighbors’ neighbors, until we encounter the target node. The number of <i>“ripples”</i> needed to reach the target provides the distance. To be specific, the identification of the shortest path between node <i>i</i> and <i>j</i> follows the following steps (<u>Gallery 2.5</u>):</p>
  
<ol>
<li><p>Start at node <i>i</i>.</p></li>

<li><p>Find the nodes directly linked to <i>i</i>. Label them distance <i>“1”</i> and put them in a queue.</p></li>

<li><p>Take the first node, labeled <i>n</i>, out of the queue (<i>n = 1</i> in the first step). Find the unlabeled nodes adjacent to it in the graph. Label them with <i>n + 1</i> and put them in the queue.</p></li>

<li><p>Repeat step <i>3</i> until you find the target node <i>j</i> or there are no more nodes in the queue.</p></li>

<li><p>The distance between <i>i</i> and <i>j</i> is the label of <i>j</i>. If <i>j</i> does not have a label, then <i>d<sub>ij</sub> = ∞</i>.</p></li>

</ol>
  
<p>The time complexity of the BFS algorithm, representing the approximate number of steps the computer needs to find <i>d<sub>ij</sub></i> on a network of <i>N</i> nodes and <i>L</i> links, is <i>O (N + L)</i>. It is linear in <i>N</i> and <i>L</i> as each node needs to be entered and removed from the queue at most once, and each link has to be tested only once.</p>
</div>

<div class="image-container" >	

<div id="second-slide">   
    <ul class="bjqs">
          <li><img src="../imgs/chapter02/ch02_12a.jpg" title="PATH: A sequence of nodes such that each node is connected to the next node along the path by a link. A path always consists of <i>n</i> nodes and <i>n - 1</i> links. The length of a path is defined as the number of its links, counting multiple edges multiple times." ></li>
          <li><img src="../imgs/chapter02/ch02_12b.jpg" title="SHORTEST PATH (geodesic path, <i>d</i>): the path with the shortest distance d between two nodes. We will call it the distance between two nodes." ></li>
          <li><img src="../imgs/chapter02/ch02_12c.jpg" title="DIAMETER (<i>d<sub>max</sub></i>): the longest shortest path in a graph, or the distance between the two furthest away nodes." ></li>
          <li><img src="../imgs/chapter02/ch02_12d.jpg" title="AVERAGE PATH LENGTH (<i>‹d›</i>): the average of the shortest paths between all pairs of nodes.<br>
CYCLE: a path with the same start and end node.<" ></li>
          <li><img src="../imgs/chapter02/ch02_12e.jpg" title="CYCLE: a path with the same start and end node.<br>
SELF-AVOIDING PATH: a path that does not intersect itself, i.e. the same node or link does not occur twice along the path." ></li>
          <li><img src="../imgs/chapter02/ch02_12f.jpg" title="SELF-AVOIDING PATH: a path that does not intersect itself, i.e. the same node or link does not occur twice along the path." ></li>
          <li><img src="../imgs/chapter02/ch02_12g.jpg" title="EULERIAN PATH: a path that traverses each link exactly once." ></li>
          <li><img src="../imgs/chapter02/ch02_12h.jpg" title="HAMILTONIAN PATH: a path that visits each node exactly once." ></li>
        </ul>
  </div>
    
    <p class="captionBold">Image 2.12</p>
    <p class="caption"><span class="imte">Pathology.</span></br>
<strong>1)</strong> PATH: A sequence of nodes such that each node is connected to the next node along the path by a link. A path always consists of <i>n</i> nodes and <i>n - 1</i> links. The length of a path is defined as the number of its links, counting multiple edges multiple times.<br>
<strong>2)</strong>  SHORTEST PATH (geodesic path, <i>d</i>): the path with the shortest distance d between two nodes. We will call it the distance between two nodes.<br>
<strong>3)</strong> DIAMETER (<i>d<sub>max</sub></i>): the longest shortest path in a graph, or the distance between the two furthest away nodes.<br>
<strong>4)</strong> AVERAGE PATH LENGTH (<i>‹d›</i>): the average of the shortest paths between all pairs of nodes.<br>
<strong>5)</strong> CYCLE: a path with the same start and end node.<br>
<strong>6)</strong> SELF-AVOIDING PATH: a path that does not intersect itself, i.e. the same node or link does not occur twice along the path.<br>
<strong>7)</strong> EULERIAN PATH: a path that traverses each link exactly once.<br>
<strong>8)</strong> HAMILTONIAN PATH: a path that visits each node exactly once.<br>
</p>
</div>


<div class="image-container" >	

<div id="banner-slide">   
    <ul class="bjqs">
          <li><img src="../imgs/chapter02/ch02_13a.jpg" title="orange node <i>0</i>"></li>
          <li><img src="../imgs/chapter02/ch02_13b.jpg" title="neighbors <i>1</i>"></li>
          <li><img src="../imgs/chapter02/ch02_13c.jpg" title="unlabeled neighbors <i>2</i>"></li>
          <li><img src="../imgs/chapter02/ch02_13d.jpg" title="...until no node is left unlabeled"></li>
        </ul>
  </div>
    
    <p class="captionBold">Image 2.13</p>
    <p class="caption"><span class="imte">The BFS algorithm applied to a small network.</span></br>
Starting from the orange node, labeled <i>”0”</i>, we identify all its neighbors, labeling them <i>”1”</i>. Then we label <i>”2”</i> the unlabeled neighbors of all nodes labeled <i>”1”</i>, and so on, in each iteration increasing the labels, until no node is left unlabeled. The length of the shortest path or the distance <i>d<sub>0i</sub></i> between node <i>0</i> and some other node <i>i</i> in the network is given by the label on node <i>i</i>. For example, the distance between node <i>0</i> and the leftmost node is <i>d<sub>03</sub> = 3</i>.</p>
</div>

</section>
<div class="pageb"></div>


<section>
<a id="Connectedness-and-Components"></br></br></a>
<h3 id="#">2.9 CONNECTEDNESS AND COMPONENTS</h3>
<p>The phone would be of limited use as a communication device if we could not call any valid phone number; the email world be rather useless if we could send emails to only certain email addresses, and not to others. From a network perspective this means that the technology behind the phone or the Internet must be capable of establishing a path between any two devices or clients, like your phone and any other phone on the network or between yours and your acquaintance’s email address. This is in fact the key utility of most networks: they are built to ensure connectedness. In this section we discuss the graph-theoretic formulation of connectedness.</p>

<p>In an undirected network two nodes <i>i</i> and <i>j</i> are connected if there is a path between them on the graph. They are disconnected if such a path does not exist, in which case we have <i>d<sub>ij</sub> = ∞</i>. This is illustrated in <u>Image 2.14a</u>, which shows a network consisting of two disconnected clusters. While there are paths between the nodes that belong to the same cluster (for example nodes <i>4</i> and <i>6</i>), there are no paths between nodes that belong to different clusters (for example nodes <i>1</i> and <i>6</i>).</p>

<div class="image-container half-width-right" >
	<img src="../imgs/chapter02/ch02_14.jpg" alt="Connected and disconnected networks." />	
    <p class="captionBold">Image 2.14</p>
    <p class="caption"><span class="imte">Connected and disconnected networks.</span></br>
(a) The network consists of two disconnected components, i.e. there is a path between any pair of nodes in the (1,2,3) component, as well in the (4,5,6,7) component. However, there are no paths between nodes that belong to different connected components. The right panel shows the adjacently matrix of the network. If the network consists of disconnected components, the adjacency matrix can be rearranged into a block diagonal form, such that all nonzero elements of the matrix are contained in square blocks along the diagonal of the matrix and all other elements are zero.<br>
(b) The addition of one link, called a <i>bridge</i>, can turn a disconnected network into a single connected component. Now there is a path between every pair of nodes in the network. Consequently the adjacency matrix cannot be written in a block diagonal form.</p>
</div>

<p>A network is connected if all pairs of nodes in the network are connected. It is disconnected if there is at least one pair with <i>d<sub>ij</sub> = ∞</i>. Clearly the network shown in Image 2.6a is disconnected, and we call its two subnetwork components (or clusters). A component is a subset of nodes in a network, so that there is a path between any two nodes that belong to the component, but one cannot add any more nodes to it that would have the same property. If a network consists of two components, a properly placed single link can connect them, making the network connected (<u>Image 2.14b</u>). Such a link is called a bridge. In general a bridge is any link that, if cut, disconnects the graph.</p>

<p>While for a small network visual inspection can help us decide if it is connected or disconnected, for a network consisting of millions of nodes connectedness is a challenging question. Several mathematical tools help us identify the connected components of a graph:</p>

<ul>
<li><p>For a disconnected network the adjacency matrix can be rearranged into a block diagonal form, such that all nonzero elements in the matrix are contained in square blocks along the matrix’ diagonal and all other elements are zero (<u>Image 2.14a</u>). Each square block will correspond to a component. We can use the tools of linear algebra to decide if the adjacency matrix is block diagonal, helping us to identify the connected components.</p></li>
<li><p>In practice, for large networks the components are more efficient identified using the breadth first search algorithm (<u>Box 2.7</u>).</p></li>
</ul>

<div class="tip">
  <h4 id="com.plex">Box 2.7</br>Finding the connected components of a graph.</h4>
  
<ol>
<li><p>Start from a randomly chosen node i and perform a BFS from this node (<u>Box 2.6</u>). Label all nodes reached this way with <i>n = 1.</i> By linking friends to each other, we obtain the <i>friendship network</i>, that plays an important role in the spread of ideas, products and habits and is of major interest to sociology, marketing and health sciences.
</p></li>
<li><p>If the total number of labeled nodes equals <i>N</i>, then the network is connected. If the number of labeled nodes is smaller than <i>N</i>, the network consists of several components. To identify them, proceed to step 3.</p></li>
<li><p>Increase the label <i>n → n + 1</i>. Choose an unmarked node <i>j</i>, label it with <i>n</i>. Use BFS to find all nodes reachable from <i>j</i>, label them with <i>n</i>. Return to step 2.</p></li>
</ol>
  
</div>

</section>
<div class="pageb"></div>


<section>
<a id="Clustering-coefficient"></br></br></a>
<h3 id="#">2.10 CLUSTERING COEFFICIENT</h3>

<p>The local clustering coefficient captures the degree to which the neighbors of a given node link to each other. For a node <i>i</i> with degree <i>k<sub>i</sub></i> the local clustering coefficient is defined as [5].</p>

<h4>\[{C_i} = \frac{{2{L_i}}}{{{k_i}\left( {{k_i} - 1} \right)}} \hspace{20 mm} (19) \]</h4>

<p>where <i>L<sub>i</sub></i> represents the number of links between the <i>k<sub>i</sub></i> neighbors of node <i>i</i>. Note that <i>C<sub>i</sub></i> is between <i>0</i> and <i>1</i>:</p>

<ul>
<li><p><i>C<sub>i</sub> = 0</i> if none of the neighbors of node <i>i</i> link to each other;</p></li>
<li><p><i>C<sub>i</sub> = 1</i> if the neighbors of node <i>i</i> form a complete graph, i.e. they all link to each other (<u>Image 2.15</u>).</p></li>
<li><p>In general <i>C<sub>i</sub></i> is the probability that two neighbors of a node link to each other: <i>C = 0.5</i> implies that there is a <i>50%</i> chance that two neighbors of a node are linked.</p></li>
<li><p>In summary <i>C<sub>i</sub></i> measures the network’s local density: the more densely interconnected the neighborhood of node <i>i</i>, the higher is <i>C<sub>i</sub></i>.</p></li>
</ul>

<p>The degree of clustering of a whole network is captured by the average <i>clustering coefficient, ‹C›</i>, representing the average of <i>C<sub>i</sub></i> over all nodes <i>i = 1, ..., N</i> [5],</p>

<h4>\[\left\langle C \right\rangle  = \frac{1}{N}\sum\limits_{i = 1}^N {{C_i}} \hspace{20 mm} (20) \]</h4>

<p>In line with the probabilistic interpretation <i>‹C›</i> is the probability that two neighbors of a randomly selected node link to each other.</p>

<p>While Eq. (19) is defined for undirected networks, the clustering coefficient can be generalized to directed and weighted [6,7,8,9]) networks as well. Note that in the network literature one also often encounters the <i>global clustering coefficient</i>, defined in Appendix A</p>

<div class="image-container" >
	<img src="../imgs/chapter02/ch02_15.jpg" alt="Clustering Coefficient." />	
    <p class="captionBold">Image 2.15</p>
    <p class="caption"><span class="imte">Clustering Coefficient.</span></br>
The local clustering coefficient, <i>C<sub>i</sub></i> , of the central node with degree <i>k<sub>i</sub>=4</i> for three different configurations of its neighborhood. The clustering coefficient measures the local density of links in a node’s vicinity. The bottom figure shows a small network, with the local clustering coefficient of a node shown next to each node. Next to the figure we also list the network’s average clustering coefficient <i>‹C›</i>, according to Eq. (20), and its global clustering coefficient <i>C</i>, declined in <u>Appendix A</u>, Eq. (21). Note that for nodes with degrees <i>k<sub>i</sub>=0,1</i>, the clustering coefficient is taken to be zero.</p>
</div>


</section>
<div class="pageb"></div>



<section>
<a id="Case-study-and-summary"></br></br></a>
<h3 id="#">2.11 CASE STUDY AND SUMMARY</h3>

<p>The purpose of the crash course in graph theory offered in this chapter was to familiarize us with some of the basic graph theoretical concepts and tools that network science uses. They define a set of elementary network characteristics, summarized in <u>Image 2.16</u>, that will serve as a language through which we can explore real networks. Yet, many of the networks we study in network science consist of hundreds to millions of nodes and links (<u>Table 2.1</u>). To explore them, we need to go beyond the small graphs discussed in <u>Image 2.16</u> and use the introduced measures to explore large networks. A glimpse of what we are about to encounter is offered in <u>Image 2.17a</u>, where we show the protein-protein interaction network of baker’s yeast, whose nodes are proteins, two proteins being connected if there is experimental evidence that they can bind (interact) to each other. The network is obviously too complex to understand its properties through a visual inspection of its wiring diagram. We therefore need to use the tools of network science to characterize its topology.</p>

<div class="image-container" >	

<div id="third-slide">   
    <ul class="bjqs">
          <li><img src="../imgs/chapter02/ch02_16a.jpg" ></li>
          <li><img src="../imgs/chapter02/ch02_16b.jpg" ></li>
          <li><img src="../imgs/chapter02/ch02_16c.jpg" ></li>
          <li><img src="../imgs/chapter02/ch02_16d.jpg" ></li>
        </ul>
  </div>
    
    <p class="captionBold">Image 2.16</p>
    <p class="caption"><span class="imte">Graphology.</span></br>
In network science we encounter many networks distinguished by some elementary property of the underlying graph. Here we summarize the most commonly encountered elementary network types, together with their basic properties, and an illustrative list of real systems that share the particular property. Note that in many real network we need to combine several of these elementary network characteristics. For example the WWW is a directed multi-graph with self-interactions. The mobile call network is directed and weighted, without self-loops.<br>

<strong>UNDIRECTED NETWORK</strong>: a network whose links do not have a predefined direction. Examples: Internet, power grid, science collaboration networks, protein interactions.<br>
<strong>SELF-INTERACTIONS</strong>: in many networks nodes do not interact with themselves, so the diagonal elements of adjacency matrix are zero, <i>A<sub>ii</sub> = 0, i =1,...,N</i>. In some systems self-interactions are allowed; in such networks, representing the fact that node i has a self-interaction. Examples: WWW, protein interactions.<br>
<strong>DIRECTED NETWORK</strong>: a network whose links have selected directions. Examples: WWW, mobile phone calls, citation network.<br>
<strong>MULTIGRAPH</strong>: in a multigraph nodes are permitted to have multiple links (or parallel links) between them. Hence <i>A<sub>ii</sub></i> can have any positive integer.<br>
<strong>WEIGHTED NETWORK</strong>: a network whose links have a predefined weight, strength or fow parameter. The elements of the adjacency matrix are <i>A<sub>ij</sub> = 0</i> if <i>i</i> and <i>j</i> are not connected, or <i>A<sub>ij</sub> = w<sub>ij</sub></i> if there is a link with weight <i>w<sub>ij</sub></i> between them. For unweighted (binary) networks, the adjacency matrix only indicates the presence (<i>A<sub>ij</sub> = 1</i>) or the absence (<i>A<sub>ij</sub> = 0</i>) of a link between two nodes. Examples: Mobile phone calls, email network.<br>
<strong>COMPLETE GRAPH</strong>: in a complete graph all nodes are connected to each other; no self-connections.<br>

</p>
</div>

<p>Let us use the measures we introduced so far to explore some basic characteristics of this network. The undirected network of <u>Image 2.17a</u> has <i>N = 2,018</i> proteins as nodes and <i>L=2,930</i> binding interactions as links. Hence the average degree, according to Eq. (7), is <i>‹k› = 2.90</i>, suggesting that a typical protein interacts with approximately two to three other proteins. Yet, this number is somewhat misleading.<br>
Indeed, the degree distribution <i>p<sub>k</sub></i> shown in <u>Image 2.17b</u> k indicates that the vast majority of nodes have only a few links. To be precise, in this network <i>69%</i> of nodes have fewer than three links, i.e. for these <i>k < ‹k›</i> . They coexist with a few highly connected nodes, or hubs, the largest having as many as <i>91</i> links. Such wide differences in node degrees is a consequence of the network’s scale-free property, characterizing many real networks. We will see that the precise shape of the degree distribution determines a wide range of network properties, from the network’s robustness to node failures to the spread of viruses.
</p>

<p>The breath-first-search algorithm helps us determine the network’s diameter, finding <i>d<sub>max</sub> = 14</i>. We might be tempted to expect wide variations in <i>d</i>, as some nodes are close to each other, others, however, may be quite far. The distance distribution (<u>Image 2.17c</u>), indicates otherwise: <i>p<sub>d</sub></i> has a prominent peak around <i>‹d› =5.61</i>, indicating that most distances are rather short, being in the vicinity of <i>‹d›</i>. Also, pd decays fast for large <i>‹d›</i>, suggesting that large distances are essentially absent. Instead, the variance of the degrees is <i>σ<sub>d</sub> = 1.64</i>, hence we have <i>d= 5.61 ± 1.64</i>, i.e. most path lengths are in the clise vicinity of <i>‹d›</i> . These are manifestations of the small world property, another common feature of real networks, indicating that most nodes are rather close to each other.</p>

<div class="image-container half-width-right" >
	<img src="../imgs/chapter02/ch02_17.jpg" alt="Characterizing a real network." />	
    <p class="captionBold">Image 2.17</p>
    <p class="caption"><span class="imte">Characterizing a real network.</span></br>
<strong>(a)</strong> The protein-protein interaction (PPI) network of yeast, a network frequently studied not only by biologists, but also by network scientists. The nodes of the network are proteins and links correspond to experimentally documented protein-protein binding interactions. The figure indicates that the network, consisting of <i>N=2,018</i> nodes and <i>L=2,930</i> links, has a giant component that connects 81% of the proteins, several smaller components, and numerous isolated proteins that do not interact with any other node.<br>
<strong>(b)</strong> The degree distribution, <i>p<sub>k</sub></i>, of the PPI network, providing the probability that a randomly chosen node has degree <i>k</i>. As <i>N<sub>k</sub>= Np<sub>k</sub></i>, the degree distribution provides the number of nodes with degree <i>k</i>. The degree distribution indicates that proteins of widely different degrees coexist in the PPI: most nodes have only a few links, a few, however, have dozens of links, representing the hubs of the network.<br>
<strong>(c)</strong> The distance distribution, pd for the PPI network, providing the probability that two randomly chosen nodes have a distance d between them (shortest path). The dotted line shows the average path length, which is <i>‹d› =5.61</i>.<br>
<strong>(d)</strong> The dependence of the average clustering coefficient on the node’s degree, <i>k</i>. The <i>C(k)</i> function is measured by averaging over the local clustering coefficient of all nodes with the same degree <i>k</i>.
</p>
</div>

<p>The breath first search algorithm will also convince us that the protein interaction network is not connected, but consists of <i>185</i> components, shown as isolated clusters in <u>Image 2.17a</u>. The largest, called the giant component, contains <i>1,647</i> of the <i>2,018</i> nodes; all other components are tiny compared to it. As we will see in the coming chapters, such fragmentation is common in real networks.
</p>

<p>The average clustering coefficient of the network is <i>‹C› =0.12</i>, which, as we will come to appreciate in the coming chapters, is rather large, indicating a significant degree of local clustering. A further caveat is provided by the dependence of the clustering coefficient on the node’s degree, or the <i>C(k)</i> function (<u>Image 2.17d</u>), which indicates that the clustering coefficient of the small nodes is significantly higher than the clustering coefficient of the hubs. This suggests that the small degree nodes are locates in dense local neighborhoods, while the neighborhood of the hubs is much more sparse. This is a consequence of network hierarchy, another widely shared network property.<br>
Finally, a visual inspection reveals an interesting pattern: hubs have a tendency to connect to small nodes, giving the network a hub and spoke character. This is a consequence of degree correlations, which influence a number of network characteristics, from the spread of ideas and viruses in social networks to the number of driver nodes needed to control a network.<br>
Taken together, <u>Image 2.17</u> illustrates that the quantities we introduced in this chapter can help us diagnose several key properties of real networks. The purpose of the coming chapters is to study systematically these network characteristics, understanding what they tell us about the behavior of a complex system.</p>

</section>
<div class="pageb"></div>


<section>
<a id="Global-clustering-coefficient"></br></br></a>
<h3 id="#">ADVANCED TOPICS: GLOBAL CLUSTERING COEFFICIENT</h3>

<p>In the network literature one often encounters the <i>global clustering coefficient</i>, which measures the total number of closed triangles in a network. Indeed, <i>L<sub>i</sub></i> in Eq. (19) is the number of triangles that node i participates in, as each link between two neighbors of node <i>i</i> closes a triangle (Image 2.15). Hence the degree of a network’s global clustering is captured by the global <i>clustering coefficient</i>, defined as</p>

<h4>\[ C = \frac{{3 \times Number\hspace{2 mm}Of\hspace{2 mm}Triangles}}{{Number\hspace{2 mm}Of\hspace{2 mm} Conneted\hspace{2 mm}Triples}} \hspace{20 mm} (21) \]</h4>

<p>where a <i>connected triplet</i> consists of three nodes that are connected by two (open triplet) or three (closed triplet) undirected links. For example, an <i>A, B, C</i> triangle is made of three triples, <i>ABC, BCA</i> and <i>CAB</i>. In contrast a chain of connected nodes <i>A, B, C</i>, in which <i>B</i> connects to <i>A</i> and <i>C</i> but <i>A</i> does not link to <i>C</i> forms a single open triplet. The factor of three in the denominator of Eq. (21) is due to the fact that each triangle is counted tree times in the triple count. The roots of the global clustering coefficient go back to the social network literature of the 1940s [10,11], hence <i>C</i> is often called the <i>number of transitive triplets</i>.</p>

<p>Note that the average clustering coefficient <i>‹C›</i> defined in (20) and the global clustering coefficient defined in (21) are not equivalent.</p>

<p>Indeed, take a network that is a double star consisting of <i>N</i> nodes, where nodes <i>1</i> and <i>2</i> are joined to each other and to all other vertices, and there are no other links. Then the local clustering coefficient <i>C<sub>i</sub></i> is <i>1</i> for <i>i ≥ 3</i> and <i>2/(N − 1)</i> for <i>i = 1, 2</i>. It follows that the average clustering coefficient of the network is <i>‹C› = 1−O(1)</i>, while the global clustering coefficient gives <i>C ~ 2/N</i>. In less extreme networks the definitions will give more comparable values, but they will still differ from each other [13]. For example, in <u>Image 2.15</u> we have <i>‹C› = 0.31</i> while <i>C = 0.375</i> .</p>

</section>
<div class="pageb"></div>

<section>
<a id="Bibliography"></br></br></a>
<a id="3" style="display: block;"></a>
<h3 id="books">BIBLIOGRAPHY</h3>
<div class="list">
	
	<ul>
		
<li><p>
[1] A.-L. Barabási and R. Albert. <a href="http://www.barabasilab.com/pubs/CCNR-ALB_Publications/199910-15_Science-Emergence/199910-15_Science-Emergence.pdf"><em>Emergence of scaling in random networks</em></a>. Science, 286(5439):509–512, 1999.
</p>
</li>
		
<li><p>
[2] G. Gilder. <a href="http://www.seas.upenn.edu/~gaj1/metgg.html"><em>Metcalfe’s law and legacy</em></a>. Forbes ASAP, 1993.
</p>
</li>

<li><p>
[3] B. Briscoe, A. <a href="http://spectrum.ieee.org/computing/networks/metcalfes-law-is-wrong"><em>Metcalfe’s law is wrong</em></a>. IEEE Spectrum, 43(7):34–39, 2006.
</p>
</li>

<li><p>
[4] K.-I. Goh, M. E. <a href="http://www.pnas.org/content/104/21/8685.full"><em>The human disease network</em></a>. Proceedings of the National Academy of Sciences, 104(21):8685– 8690, 2007.
</p>
</li>

<li><p>
[5] D. J. Watts and S. H. Strogatz. <a href="http://www.nature.com/nature/journal/v393/n6684/abs/393440a0.html"><em>Collective dynamics of ‘small-world’ networks</em></a>. Nature, 393(6684):440–442, 1998.
</p>
</li>

<li><p>
[6] A. Barrat, M. Barthélemy, R. Pastor-Satorras, and A. Vespignani. <a href="http://www.pnas.org/content/101/11/3747"><em>The architecture of complex weighted networks</em></a>. Proceedings of the National Academy of Sciences of the United States of America, 101(11):3747–3752, 2004.
</p>
</li>

<li><p>
[7] J. P. Onnela, J. Saramäki, J. Kertész, and K. Kaski. <a href="http://lib.tkk.fi/Diss/2006/isbn9512282704/article6.pdf"><em>Intensity and coherence of motifs in weighted complex networks</em></a>. Phys. Rev. E, 71:065103, 2005.
</p>
</li>

<li><p>
[8] B. Zhang and S. Horvath. <a href="http://labs.genetics.ucla.edu/horvath/GeneralFramework/WeightedNetwork2005.pdf"><em>A general framework for weighted gene coexpression network analysis</em></a>. Statistical Applications in Genetics and Molecular Biology, 4:17, 2005.
</p>
</li>

<li><p>
[9] P. Holme, S. M. Park, J. B. Kim, and C. R. Edling. <a href="http://arxiv.org/pdf/cond-mat/0411634.pdf"><em>Korean university life in a network perspective: Dynamics of a large affiliation network</em></a>. Physica A: Statistical Mechanics and its Applications, 373(0):821–830, 2007.
</p>
</li>

<li><p>
[10] R. D. Luce and A. D. Perry. <a href="http://aris.ss.uci.edu/~lin/50.pdf"><em>A method of matrix analysis of group structure</em></a>. Psychometrika, 14:95–116, 1949.
</p>
</li>

<li><p>
[11] S. Wasserman and K Faust. <a href="http://train.ed.psu.edu/WFED-543/SocNet_TheoryApp.pdf"><em>Social Network Analysis: Methods and Applications</em></a>. Cambridge University Press, 1994.
</p>
</li>

<li><p>
[12] Y.-Y. Ahn, S. E. Ahnert, J. P. Bagrow, A.-L. Barabási. <a href="http://www.nature.com/srep/2011/111215/srep00196/full/srep00196.html"><em>Flavor network and the principles of food pairing</em></a>. Scientific Reports 196, 2011.
</p>
</li>

<li><p>
[13] B. Bollobás and O. M. Riordan. <a href="http://hydra.icgeb.trieste.it/~pongor/biophys-homepage/Network_handbook/CH001.pdf"><em>Mathematical results on scale-free random graphs</em></a>. in Stefan Bornholdt, Hans Georg Schuster, <i>Handbook of Graphs and Networks: From the Genome to the Internet</i>, (2003 Wiley-VCH Verlag GmbH & Co. KGaA).
</p>
</li>

<li><p>
[14] K.-I. Goh, M. E. Cusick, D. Valle, B. Childs, M. Vidal, and A.-L. Barabási. <a href="http://ccsb.dfci.harvard.edu/web/export/sites/default/ccsb/publications/papers/2007/Goh-etal_Diseasome_PNAS_07.pdf"><em>The human disease network</em></a>. Proceedings of the National Academy of Sciences 104:21, 8685 (2007).
</p>
</li>


</ul>
</div>
<div style="page-break-after:always;"> </div>
</section>


</div>
</div>  
  
  
  
  



  
  
  
  
  




	<div id="bottom">
		<div id="footer">
        
        
			<div class="one-third" id="licenses">
        <h4>Licenses</h4>
        <p>
          <a class="license-badge" rel="license" href="http://creativecommons.org/licenses/by-nc/3.0/"><img alt="Creative Commons License" style="border-width:0" src="http://i.creativecommons.org/l/by-nc/3.0/88x31.png" /></a>
         
        </p>

        <p>
          The book's text and illustrations are licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc/3.0/">Creative Commons Attribution-NonCommercial 3.0 Unported License</a>.
        </p>
      </div>
      

      <div class="one-third">
        <h4>Credit</h4>

        <p>Design & Data Visualization: <a href="http://www.mamartino.com">Mauro Martino</a>, Márton Pósfai, Gabriele Musella</p>

      </div>

      <div class="one-third">
        <h4>Author</h4>
        <p><a href="http://www.barabasi.com">ALBERT-LÁSZLÓ BARABÁSI</a> is director of the <a href="http://barabasilab.neu.edu">Center for Complex Network Research</a> at Northeastern University.</p>

        <p>He is the author of <a href="http://barabasilab.com/LinkedBook/">Linked</a>.</p>

        <p><a href="https://twitter.com/barabasi">BARABÁSI in Twitter</a> </p>
      </div>
		</div>
	</div>
</body>
</html>